railties/lib/rails/generators/named_base.rb

```diff
@@ -98,16 +98,18 @@ def index_helper # :doc:
           uncountable? ? "#{plural_route_name}_index" : plural_route_name
         end
 
-        def show_helper # :doc:
-          "#{singular_route_name}_url(@#{singular_table_name})"
+        def show_helper(local: false, suffix: "url", entity: nil) # :doc:
+          instance_var = local ? "" : "@"
+          entity ||= "#{instance_var}#{singular_table_name}"
+          "#{singular_route_name}_#{suffix}(#{entity})"
         end
 
         def edit_helper # :doc:
           "edit_#{show_helper}"
         end
 
-        def new_helper # :doc:
-          "new_#{singular_route_name}_url"
+        def new_helper(suffix: "url") # :doc:
+          "new_#{singular_route_name}_#{suffix}"
```

Comment:
If we do change the name of the `show_helper` args, this should match:

```suggestion
        def new_helper(route_suffix: "url") # :doc:
          "new_#{singular_route_name}_#{route_suffix}"
```